app/Database/Tables.hs

```diff
@@ -224,23 +219,38 @@ instance FromJSON SvgJSON
 -- JSON encoding/decoding
 instance FromJSON Courses where
   parseJSON = withObject "Expected Object for Courses" $ \o -> do
-    newCode <- o .:? "code" .!= "CSC???"
-    newTitle  <- o .:? "courseTitle"
-    newDescription  <- o .:? "courseDescription"
-    newPrereqString <- o .:? "prerequisite"
-    let newPrereqs = fmap (T.pack . show . parseReqs . T.unpack) newPrereqString
-    newExclusions <- o .:? "exclusion"
-    newCoreqs <- o .:? "corequisite"
-    return $ Courses newCode
-                     newTitle
-                     newDescription
-                     newPrereqs
-                     newExclusions
-                     Nothing -- breadth
-                     Nothing -- distribution
-                     Nothing -- (Just prereqString)
-                     newCoreqs
-                     []
+    codeExtraChars <- o .:? "code" .!= "CSC?????"
```

Comment:
This isn't great, we don't want `CSC???` to appear in the database anywhere. I'd just do `o .: "code"` and let the parsing fail if the key is missing.